plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/project/ManageProjectsWizardPageModel.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.wizard.project;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.jboss.tools.common.ui.databinding.ObservableUIPojo;
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistrySingleton;
+import org.jboss.tools.openshift.common.core.connection.IConnection;
+import org.jboss.tools.openshift.common.core.connection.IConnectionsRegistryListener;
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.core.connection.ConnectionProperties;
+
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IProject;
+
+/**
+ * @author jeff.cantrill
+ */
+public class ManageProjectsWizardPageModel extends ObservableUIPojo {
+
+	public static final String PROPERTY_SELECTED_PROJECT = "selectedProject";
+	public static final String PROPERTY_PROJECTS = "projects";
+
+	private Connection connection;
+	private IProject selectedProject;
+	private List<IProject> projects;
+	private IConnectionsRegistryListener connectionChangeListener;
+
+	public ManageProjectsWizardPageModel(IProject project, Connection connection) {
+		this(connection);
+		setSelectedProject(project);
+	}
+
+	public ManageProjectsWizardPageModel(Connection connection) {
+		this.connection = connection;
+		this.connectionChangeListener = onConnectionsChanged();
+		ConnectionsRegistrySingleton.getInstance().addListener(connectionChangeListener );
+	}
+
+	private IConnectionsRegistryListener onConnectionsChanged() {
+		return new IConnectionsRegistryListener() {
+		
+			@Override
+			public void connectionRemoved(IConnection connection) {
+				if(connection != null && connection.equals(ManageProjectsWizardPageModel.this.connection)){
+					ManageProjectsWizardPageModel.this.connection = null;
+					setProjects(Collections.<IProject>emptyList());
+				}
+			}
+			
+			@SuppressWarnings("unchecked")
+			@Override
+			public void connectionChanged(IConnection connection, String property, Object oldValue, Object newValue) {
+
+				if(connection != null && connection.equals(ManageProjectsWizardPageModel.this.connection) && ConnectionProperties.PROPERTY_PROJECTS.equals(property)){
+					setProjects((List<IProject>) newValue); 
+				}
+			}
+			
+			@Override
+			public void connectionAdded(IConnection connection) {
+				if(connection != null && connection.equals(ManageProjectsWizardPageModel.this.connection)){
+					ManageProjectsWizardPageModel.this.connection = (Connection) connection;
+					loadProjects();
+				}
+			}
+		};
+	}
+
+	public void loadProjects() {
+		if (connection == null) {
+			setProjects(Collections.<IProject>emptyList());
+		} else {
+			List<IProject> projects = connection.getResources(ResourceKind.PROJECT);
+			setProjects(projects);
```

Comment:
ok so sounds fully sufficient, it was just not clear to me when only reading code (and not live testing).